pyprof/prof/convert.py

```diff
@@ -15,66 +15,55 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from collections import OrderedDict
-from .utility import Utility
 from .base import OperatorLayerBase
-
+from .tensor import Tensor
 
 class Convert(OperatorLayerBase):
     """
 	Class to handle convert operations.
 	"""
-    ops = ["byte", "char", "double", "float", "half", "int", "long", "short", "to"]
+    ops = ["byte", "char", "double", "float", "half", "int", "long",
+           "short", "to"]
 
     def __init__(self, d):
         marker = eval(d.argMarker[0])
         mod = marker['mod']
         op = marker['op']
         args = marker['args']
 
-        self.marker = marker
-        self.mod_ = mod
-        self.op_ = op
-        self.args = args
+        self._mod = mod
+        self._op = op
 
         assert (mod == "Tensor")
         assert (op in Convert.ops)
         assert (len(args) == 1)
 
-        #The argument could be a tensor or scalar
         t = args[0]
         if t['type'] == "tensor":
-            shape = t['shape']
-            stype = t['dtype']
-        else:
-            shape = (1, )
-            stype = t['type']
-        if self.op_ == "to":
-            op = stype
+            self.inp = Tensor(t['shape'], t['dtype'])
+        else: # scalar
+            self.inp = Tensor([], t['type'])
 
-        self.shape = shape
-        self.stype = stype
-        self.dtype = op
+        if op == "to":
+            # make the output the same as input
+            self.out = self.inp
```

Comment:
Oh, `inp` is `input`! This is the first time I have seen this reference. Can you see how this would be confusing? What advantage do you get from leaving out `ut` in the name? Please rename all `inp` to `input` to make it clear what the variable and intention are.

Reply:
Should consider renaming `out` to `output` for consistency as well

Reply:
Also, the comment is obvious and not needed based on reading the code. If you need a comment, use it to explain why you are setting output to input.